content/posts/gateway-vs-ingress-api.md

```diff
@@ -0,0 +1,266 @@
+
+## Custom CRDs > Ingress API
+
+Being stuck with annotations also sacrifice the usability of the Ingress controllers.
+
+Controllers therefore solved the limitations of the Ingress API by creating their [own custom resources](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/). The example below shows configuring Ingress to route traffic between two versions of an application using APISIX's custom resource:
+
+```yaml {title="apisix-ingress-manifest.yaml"}
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixRoute
+metadata:
+  name: api-routes
+spec:
+  http:
+    - name: route-1
+      match:
+        hosts:
+          - local.navendu.me
+        paths:
+          - /v1
+      backends:
+        - serviceName: bare-minimum-api-v1
+          servicePort: 8080
+    - name: route-2
+      match:
+        hosts:
+          - local.navendu.me
+        paths:
+          - /v2
+      backends:
+        - serviceName: bare-minimum-api-v2
+          servicePort: 8081
+```
+
+These CRDs made it much easier to configure Ingress, but you are tied to the specific Ingress control implementation. Without the Ingress API evolving, you had to choose between usability or portability.
+
+## Extending Ingress and Evolution to Gateway API
+
+Ingress API was not broken; it was limited. The Gateway API was designed to overcome these limitations.
+
+{{< blockquote author="gateway-api.sigs.k8s.io" link="[Link](https://gateway-api.sigs.k8s.io/)" title="What is the Gateway API?" >}}
+(Gateway API) aim to evolve Kubernetes service networking through expressive, extensible, and role-oriented interfaces ...
+{{< /blockquote >}}
+
+It takes inspiration from the custom CRDs of different Ingress controllers mentioned earlier.
+
+The Gateway API adds [many features](https://gateway-api.sigs.k8s.io/#gateway-api-concepts) "on top" of the Ingress API's capabilities. This includes HTTP header-based matching, weighted traffic splitting, and other features that require custom proprietary annotations with the Ingress API.
+
+Traffic split with APISIX Ingress resource (see [ApisixRoute/v2beta3 reference](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2beta3/)):
+
+```yaml {title="apisix-ingress-manifest.yaml"}
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixRoute
+metadata:
+  name: traffic-split
+spec:
+  http:
+    - name: rule-1
+      match:
+        hosts:
+          - local.navendu.me
+        paths:
+          - /get*
+      backends:
+        - serviceName: bare-minimum-api-v1
+          servicePort: 8080
+          weight: 90
+        - serviceName: bare-minimum-api-v2
+          servicePort: 8081
+          weight: 10
+```
+
+Traffic split with Gateway API (see [Canary traffic rollout](https://gateway-api.sigs.k8s.io/guides/traffic-splitting/#canary-traffic-rollout)):
+
+```yaml {title="kubernetes-gateway-manifest.yaml"}
+apiVersion: gateway.networking.k8s.io/v1beta1
```

Comment:
https://apisix.apache.org/docs/ingress-controller/aeps/gateway-api/

In APISIX Ingres, only `v1alpha2` is supported, and `v1beta1` does not support.